sdk/device.go

```diff
@@ -299,6 +305,36 @@ func NewDeviceFromConfig(
 	return d, nil
 }
 
+// parseContext is a utility function to parse the context map and render any templates
+// which may exist in the context values.
+func parseContext(ctx map[string]string) error {
+	ctxTmpl := template.New("ctx").Funcs(template.FuncMap{
+		"env": os.Getenv,
+	})
+
+	for k := range ctx {
+		val := ctx[k]
+		if val == "" {
+			continue
+		}
+
+		tmpl, err := ctxTmpl.Parse(val)
+		if err != nil {
+			return err
+		}
+		buf := bytes.Buffer{}
+		if err := tmpl.Execute(&buf, val); err != nil {
+			return err
+		}
+
+		// NOTE: The SDK does not currently verify that the specified environment
```

Comment:
would it be helpful to log a debug message when thats the case?

Reply:
good call -- updated with logging!